compiler/rustc_resolve/src/imports.rs

```diff
@@ -1371,9 +1369,9 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
             self.lint_buffer.buffer_lint_with_diagnostic(
                 UNUSED_IMPORTS,
                 id,
-                import.span,
+                import.use_span,
                 format!("the item `{source}` is imported redundantly"),
-                BuiltinLintDiag::RedundantImport(redundant_spans, source),
+                BuiltinLintDiag::RedundantImport(redundant_spans, source, import.use_span),
```

Comment:
`import.use_span` cannot be used for removal in general case (e.g. if the redundant import in inside braces `use a::{b, c};`).

Ideally, the removal algorithm for usual unused imports from `fn check_unused` should be used, but otherwise maybe it can be approximated with `import.span/use_span/root_span` somehow.

Reply:
Done with some code changes, I change `check_for_redundant_imports` to return spans, and add a new `ImportFinderVisitor` to find UseTree, and then use `calc_unused_spans` to fix spans.